Automattic/wp-calypso
# WPORG Plugins: use 1.2 endpoint and pass locale parameter.

#### Changes proposed in this Pull Request

* uses latest `https://api.wordpress.org/plugins/info/1.2/` instead of `https://api.wordpress.org/plugins/info/1.1/`. The new endpoint has more data like "active installations" and also accepts only GET requests
* pass the locale depending on current user locale in both plugins list and plugin details requests
* refactor away the specific `pluginRequest` and use `getRequest` 
* fetches only data needed

#### Testing instructions



* on a language different than English visit:
* plugins list `/plugins`, perform a search, plugin details
* make sure that the results are translated. The featured list on the top is coming from WPCOM and thus only available in English. Some plugins may not have translations in the language selected

|Before | After|
|-------|------|
|<img width="1096" alt="SS 2021-11-05 at 15 15 00" src="https://user-images.githubusercontent.com/12430020/140515898-9c1c60ff-7ecf-4919-a668-4dc6c4c56540.png">|<img width="1084" alt="SS 2021-11-05 at 15 14 31" src="https://user-images.githubusercontent.com/12430020/140515845-caf2252e-3f7a-4bd4-82d2-67280fe9e979.png">|



Fixes #57635
